### ISSUE TITLE
wrong value of MemWrite signal?

### ISSUE THREAD

**Issue by aaarbk** [arm-university/Graphical-Micro-Architecture-Simulator#8]
I assembled and executed a program with the lone instruction
`STUR	X1, [X2, #20]`

In the graphical output, the `MemRead` signal is 1 and the `MemWrite` is 0. Shouldn't `MemWrite` be 1 here?

![](https://i.imgur.com/OZeFee6.png)

**Comment by simdeistud** [comment]
> Shouldn't `MemWrite` be 1 here?

Yes @aaarbk , this is a bug in the ControlUnitConfiguration and the SingleCycleVisualization as you can see from [ this commit](https://github.com/simdeistud/LEGv8-Simulator/commit/eefbb281440a446b3f2dc030518f392667f960f0) I just made to my forked version. This repository is basically abandoned, if you wish to use a fixed and more complete version of this simulator you can check out my fork here: https://github.com/simdeistud/LEGv8-Simulator and you can download a working release.
